core/src/main/java/org/arquillian/smart/testing/scm/git/GitChangeResolver.java

```diff
@@ -59,23 +59,29 @@ public GitChangeResolver(File projectDir, Scm scm) {
     public GitChangeResolver(File dir, String previous, String head) {
         this.previous = previous;
         this.head = head;
-        final FileRepositoryBuilder builder = new FileRepositoryBuilder();
-        try {
-            this.git = new Git(builder.readEnvironment().findGitDir(dir).build());
-        } catch (IOException e) {
-            throw new IllegalArgumentException("Unable to find git repository for path " + dir.getAbsolutePath(), e);
+        final FileRepositoryBuilder fileRepositoryBuilder = getFileRepositoryBuilder();
+        if (fileRepositoryBuilder.getGitDir() != null) {
+            try {
+                this.git = new Git(fileRepositoryBuilder.build());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            this.repoRoot = git.getRepository().getDirectory().getParentFile();
+        } else {
+            logger.warn("Unable to find git repository for path " + dir.getAbsolutePath());
```

Comment:
How about throwing an exception when any strategy using scm information is used....?

Reply:
We can throw, but in that case user has to use disable flag to run even `mvn clean` because it's failing while registering extension.

So it's unnecessary burdon for user to use disable flag for mvn clean something like `mvn clean -Dsmart.testing.disable` to make clean runnable

Reply:
Yeah, that's why I said "when any strategy using scm information is used"

Reply:
How about "if user is using any strategy using scm information, then don't add any dependency related to that & warn user to use this strategy you need scm in place." this?

Reply:
well, I think that it should fail, because the ST cannot work properly. At least this is what I would expect as a user.
So, here - at this line - I would warn: `"Unable to find a git repository for the path %s - some strategies won't work properly. Make sure it is a git repository."`
Then, when any strategy tries to get the scm information using method [diff](https://github.com/arquillian/smart-testing/pull/221/files#diff-f1255f2030b502bc7deffb7a6797da92R83) then check if the git is set otherwise throw an exception.